lib/index.js

```
@@ -3611,6 +3611,79 @@ Stream.prototype.merge = function () {
 exposeMethod('merge');
 
 /**
+ * Takes a Stream of Streams and merges their values and errors into a
+ * single new Stream, limitting the number of unpaused streams that can
+ * running at any one time.
+ *
+ * Note that no guarantee is made with respect to the order in which
+ * values for each stream end up in the merged stream. Values in the
+ * merged stream will, however, respect the order they were emitted from
+ * their respective streams.
+ *
+ * @id mergeLimit
+ * @section Higher-order Streams
+ * @name Stream.mergeWithLimit()
+ * @api public
+ *
+ * var txt = _(['foo.txt', 'bar.txt']).flatMap(readFile)
+ * var md = _(['baz.md']).flatMap(readFile)
+ * var js = _(['bosh.js']).flatMap(readFile)
+ *
+ * _([txt, md, js]).mergeWithLimit(2);
+ * // => contents of foo.txt, bar.txt, baz.txt and bosh.js in the order
+ * they were read, but bosh.js is not read until either foo.txt and bar.txt
+ * has completely been read or baz.txt has been read
+ */
+
+
+Stream.prototype.mergeWithLimit = function (n){
+    var self = this;
+    var processCount = 0;
+    var waiting = false;
+    n = typeof n != 'number' ? Infinity : n;
+    if (n === Infinity) {
+        return this.merge();
+    } else if (n < 1) {
+        return _([]);
```

This should throw an error instead of returning an empty stream.